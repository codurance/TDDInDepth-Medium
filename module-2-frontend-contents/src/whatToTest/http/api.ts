import * as axios from "axios";
import {NewProduct} from "../types/NewProduct.ts";
import {Product} from "../types/Product.ts";
import {generateId} from "./generateId.ts";


const request = axios.default.create({
    baseURL: "http://localhost:5000",
})

export const createProductRequest = async (product: NewProduct): Promise<Product> => {
    const {data} = await request.post<Product>("/products", {
        id: generateId(),
        ...product,
    });

    return data;
}

export const getProductsRequest = async (): Promise<Product[]> => {
    const {data} = await request.get<Product[]>("/products");

    return data;
}