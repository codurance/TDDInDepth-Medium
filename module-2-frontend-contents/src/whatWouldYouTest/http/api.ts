import * as axios from "axios";
import {Todo} from "../types/Todo.ts";
import {NewTodo} from "../types/NewTodo.ts";
import {generateId} from "./generateId.ts";

const request = axios.default.create({
    baseURL: "http://localhost:5000",
})

export const getTodosRequest = async () => {
    const {data} = await request.get<Todo[]>("/todos");

    return data;
}

export const updateTodoRequest = async (todo: Todo) => {
    await request.patch(`/todos/${todo.id}`, todo);
}

export const createTodoRequest = async (newTodo: NewTodo) => {
    const todo: Todo = {id: generateId(), status: "todo", ...newTodo,};
    const {data} = await request.post<Todo>("/todos", todo);

    return data;
}