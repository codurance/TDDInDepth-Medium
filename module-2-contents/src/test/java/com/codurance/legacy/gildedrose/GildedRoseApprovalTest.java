package com.codurance.legacy.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

@UseReporter(DiffReporter.class)
public class GildedRoseApprovalTest {

    @Test
    public void shouldDecreaseTheItemQualityWhenQualityIsUpdated() {

        Item[] items = new Item[] { new Item("foo", 0, 0) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("Items", items);
    }


    @Test
    public void shouldDecreaseTheItemQualityWhenQualityIsUpdatedDuringThirtyDays() {

        ByteArrayOutputStream fakeoutput = new ByteArrayOutputStream();
        System.setOut(new PrintStream(fakeoutput));
        System.setIn(new ByteArrayInputStream("a\n".getBytes()));

        Program.main();
        String output = fakeoutput.toString();

        Approvals.verify(output);
    }
}
