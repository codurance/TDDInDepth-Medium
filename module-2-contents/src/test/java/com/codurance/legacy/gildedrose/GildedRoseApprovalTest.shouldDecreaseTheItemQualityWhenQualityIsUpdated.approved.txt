Items[0] = foo, -1, 0
